trusted_shuffler/trusted_shuffler/src/lib.rs

```diff
@@ -33,45 +33,77 @@ use std::{
 use tokio::{sync::oneshot, time::Duration};
 
 #[derive(Debug, PartialEq)]
-pub struct TrustedShufflerRequest {
+pub struct EncryptedRequest {
+    // TODO(#3282) Unify `body` and `data` from Request and Response.
     pub body: Vec<u8>,
     pub headers: hyper::HeaderMap,
     pub uri: Uri,
 }
 
-impl TrustedShufflerRequest {
-    // We only compare the body, and we need revisit this and decide if this
-    // is how we want to sort our requests.
-    fn compare_body(&self, r2: &TrustedShufflerRequest) -> Ordering {
+// TODO(#3281): Add decryption.
+impl EncryptedRequest {
+    fn decrypt(self) -> PlaintextRequest {
+        PlaintextRequest {
+            body: self.body,
+            headers: self.headers,
+            uri: self.uri,
+        }
+    }
+}
+
+pub struct PlaintextRequest {
+    pub body: Vec<u8>,
+    pub headers: hyper::HeaderMap,
+    pub uri: Uri,
+}
+
+impl PlaintextRequest {
+    // We need to sort the plain requests.
+    fn compare_body(&self, r2: &PlaintextRequest) -> Ordering {
         self.body.cmp(&r2.body)
     }
 }
 
 #[derive(Debug, PartialEq)]
-pub struct TrustedShufflerResponse {
+pub struct EncryptedResponse {
     pub headers: hyper::HeaderMap,
     pub data: hyper::body::Bytes,
     pub trailers: hyper::HeaderMap,
 }
 
-impl TrustedShufflerResponse {
-    fn empty() -> TrustedShufflerResponse {
-        TrustedShufflerResponse {
+pub struct PlaintextResponse {
+    pub headers: hyper::HeaderMap,
+    pub data: hyper::body::Bytes,
+    pub trailers: hyper::HeaderMap,
+}
+
+impl PlaintextResponse {
+    fn empty() -> PlaintextResponse {
+        PlaintextResponse {
             headers: hyper::HeaderMap::new(),
             data: hyper::body::Bytes::new(),
             trailers: hyper::HeaderMap::new(),
         }
     }
+
+    // TODO(#3281): Add encryption.
+    fn encrypt(self) -> EncryptedResponse {
```

Comment:
Encrypting would require a key that will be stored by an encryptor object provided by the attestation crate. So not sure if the `encrypt` function could be a member of the `PlaintextResponse`

Reply:
Yeah, I will need to revisit that once I implement #3281 .